corehq/apps/users/user_data.py

```diff
@@ -60,12 +67,34 @@ def save(self):
 
     @property
     def _provided_by_system(self):
-        return {
+        provided_data = {
             **(self.profile.fields if self.profile else {}),
             PROFILE_SLUG: self.profile_id or '',
             COMMCARE_PROJECT: self.domain,
         }
 
+        def _add_location_data():
+            if self._couch_user.get_location_id(self.domain):
+                provided_data[COMMCARE_LOCATION_ID] = self._couch_user.get_location_id(self.domain)
+                provided_data[COMMCARE_PRIMARY_CASE_SHARING_ID] = self._couch_user.get_location_id(self.domain)
+
+            if self._couch_user.get_location_ids(self.domain):
+                provided_data[COMMCARE_LOCATION_IDS] = user_location_data(
+                    self._couch_user.get_location_ids(self.domain))
+
+        if settings.UNIT_TESTING:
+            # Some test don't have an actual user existed
+            if self._couch_user:
+                _add_location_data()
+        else:
+            _add_location_data()
```

Comment:
nit: you can combine these conditionals and avoid needing to pull the body out like so:
```suggestion
        # Some test don't have an actual user existed
        if self._couch_user or not settings.UNIT_TESTING:
            _add_location_data()
```